src/models/vulnerabilityRating.ts

```diff
@@ -0,0 +1,56 @@
+/*!
+This file is part of CycloneDX JavaScript Library.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+Copyright (c) OWASP Foundation. All Rights Reserved.
+*/
+
+import { ScoreMethod, Severity } from '../enums'
+import { Comparable, SortableSet } from '../helpers/sortableSet'
+import { VulnerabilitySource } from './vulnerabilitySource'
+
+interface OptionalProperties {
+  source?: VulnerabilityRating['source']
+  score?: VulnerabilityRating['score']
+  severity?: VulnerabilityRating['severity']
+  method?: VulnerabilityRating['method']
+  vector?: VulnerabilityRating['vector']
+  justification?: VulnerabilityRating['justification']
+}
+
+export class VulnerabilityRating implements Comparable {
+  source?: VulnerabilitySource
+  score?: number
+  severity?: Severity
+  method?: ScoreMethod
+  vector?: string
+  justification?: string
+
+  constructor (op: OptionalProperties = {}) {
+    this.source = op.source
+    this.score = op.score
+    this.severity = op.severity
+    this.method = op.method
+    this.vector = op.vector
+    this.justification = op.justification
+  }
+
+  compare (other: VulnerabilityRating): number {
+    return (this.score ?? 0) - (other.score ?? 0)
```

Comment:
this appears to be like an insufficient comparison.
if this is enough to properly compare, then please add a in-code comment that explains why.